src/js/player.js

```diff
@@ -1620,30 +1618,29 @@ class Player extends Component {
    *         return undefined.
    */
   play() {
-    if (this.changingSrc_) {
-      this.ready(function() {
-        const retval = this.techGet_('play');
 
-        // silence errors (unhandled promise from play)
-        if (retval !== undefined && typeof retval.then === 'function') {
-          retval.then(null, (e) => {});
-        }
-      });
+    // Given a return value for a tech's `play()` method, silences "unhandled
+    // promise from play" errors.
+    const silencePromise = (value) => {
+      if (isPromise(value)) {
+        value.then(null, (e) => {});
+      }
+    };
+
+    // If the player is not ready, queue up a call to the tech's `play()`
+    // method for when it _is_ ready.
+    if (!this.isReady_) {
+      this.ready(() => silencePromise(this.techGet_('play')));
 
-    // Only calls the tech's play if we already have a src loaded
-    } else if (this.isReady_ && (this.src() || this.currentSrc())) {
+    // If the player is ready and there is a source, call the tech's `play()`
+    // method.
+    } else if (this.src() || this.currentSrc()) {
       return this.techGet_('play');
-    } else {
-      this.ready(function() {
-        this.tech_.one('loadstart', function() {
-          const retval = this.play();
 
-          // silence errors (unhandled promise from play)
-          if (retval !== undefined && typeof retval.then === 'function') {
-            retval.then(null, (e) => {});
-          }
-        });
-      });
+    // Finally, if the player is ready, but we don't have a source, wait for
+    // one to be set.
+    } else {
+      this.ready(() => this.tech_.one('loadstart', () => silencePromise(this.play())));
```

Comment:
This will fail because the function gets called with the context of the player and not the tech as relied on previously.

Reply:
D'oh.